repository/repository_test.go

```diff
@@ -61,31 +68,37 @@ func (s *FilesystemSuite) cleanUpTempDirectories() {
 
 func (s *FilesystemSuite) Test() {
 	fsPairs := []*fsPair{
-		{"mem to mem", memfs.New(), memfs.New()},
-		{"mem to os", memfs.New(), s.newFilesystem()},
-		{"os to mem", s.newFilesystem(), memfs.New()},
-		{"os to os", s.newFilesystem(), s.newFilesystem()},
+		{"mem to mem", NewLocalCopier(memfs.New()), memfs.New()},
+		{"mem to os", NewLocalCopier(memfs.New()), s.newFilesystem()},
+		{"os to mem", NewLocalCopier(s.newFilesystem()), memfs.New()},
+		{"os to os", NewLocalCopier(s.newFilesystem()), s.newFilesystem()},
+		{"os to HDFS", NewHDFSCopier(hdfsURL, s.newTempPath()), s.newFilesystem()},
+		{"mem to HDFS", NewHDFSCopier(hdfsURL, s.newTempPath()), memfs.New()},
```

Comment:
actually, the time is spent in download hdfs from internet. But I can remove this test anyways.